src/modules/DialogOverlay.vue

```diff
@@ -129,54 +130,88 @@ export default defineComponent({
         productContents.push({
           key: 'productPrice',
           name: 'Purchase Amount',
-          value: `$${parseFloat(`${this.productPrice}`).toFixed(2)}`,
+          value: `${parseFloat(`${this.productPrice}`).toFixed(2)}`,
         })
       }
       return productContents
     },
     filterContentByLang(content: ContentsProps[]): ContentsProps[] {
-      let displayedContent = undefined
+      let filteredContent = undefined
       switch (this.lang) {
         case LanguageCodeEnum.Australia:
-          displayedContent = content.filter(item =>
+          filteredContent = content.filter(item =>
             this.fieldBreakdownAu.includes(item.key)
           )
-          if (!displayedContent)
+          if (!filteredContent)
             throw new Error(
               `No content that matches AU breakdown : ${this.fieldBreakdownAu}`
             )
-          return displayedContent
+          return filteredContent
         case LanguageCodeEnum.NewZealand:
-          displayedContent = content.filter(item =>
+          filteredContent = content.filter(item =>
             this.fieldBreakdownNz.includes(item.key)
           )
-          if (!displayedContent)
+          if (!filteredContent)
             throw new Error(
               `No content that matches NZ breakdown : ${this.fieldBreakdownNz}`
             )
-          return displayedContent
+          return filteredContent
         default:
           return content
       }
     },
     orderContent(unorderedContent: ContentsProps[]): ContentsProps[] {
       switch (this.lang) {
-        case LanguageCodeEnum.Australia:
-          return unorderedContent.sort(
+        case LanguageCodeEnum.Australia: {
+          const orderedContent = unorderedContent.sort(
             (a, b) =>
               this.fieldBreakdownAu.indexOf(a.key) -
               this.fieldBreakdownAu.indexOf(b.key)
           )
-        case LanguageCodeEnum.NewZealand:
-          return unorderedContent.sort(
+          return orderedContent
+        }
+        case LanguageCodeEnum.NewZealand: {
+          const orderedContent = unorderedContent.sort(
             (a, b) =>
               this.fieldBreakdownNz.indexOf(a.key) -
               this.fieldBreakdownNz.indexOf(b.key)
           )
+          return orderedContent
+        }
         default:
           return unorderedContent
       }
     },
+    appendUnits(content: ContentsProps[]): ContentsProps[] {
+      const contentWithUnit = content.map(breakdownItem => {
+        const nameLabelPair = productLanguage.find(
+          item => item.name === breakdownItem.key
+        )
+        let valueWithUnit = ''
+        if (nameLabelPair) {
+          switch (nameLabelPair.unit.toLowerCase()) {
+            case '$':
+              valueWithUnit = `${nameLabelPair.unit}${parseFloat(
+                breakdownItem.value
+              ).toFixed(2)}`
```

Comment:
maybe this currency formatting `parseFloat(value).toFixed(2)` should be in a utility function since you're using it in a couple of places

Reply:
created a function called `convertToCurrencyFormat`